src/composition-engine.js

```diff
@@ -85,18 +116,26 @@ export class CompositionEngine {
 
   _createControllerAndSwap(context) {
     function swap(controller) {
-      return Promise.resolve(context.viewSlot.removeAll(true)).then(() => {
-        if (context.currentController) {
-          context.currentController.unbind();
-        }
-
-        context.viewSlot.add(controller.view);
-
-        if (context.compositionTransactionNotifier) {
-          context.compositionTransactionNotifier.done();
-        }
-
-        return controller;
+      let swapStrategy;
```

Comment:
Is there a reason for the assignment two lines below?
The code style around is to `let` and assign on the same line.